x-pack/plugins/data_enhanced/public/search/ui/search_session_indicator/search_session_indicator.tsx

```diff
@@ -38,6 +39,9 @@ export interface SearchSessionIndicatorProps {
   saveDisabledReasonText?: string;
 
   onOpened?: (openedState: SearchSessionState) => void;
+
+  searchSessionName?: string;
+  saveSearchSessionName?: (newName: string) => Promise<unknown>;
```

Comment:
> Why is this unknown?

I'd probably also use `any` but went for `unknown because we are tracking `any` usage. 

In this case, by `unknown, I mean that API doesn't care what this promise returns because it isn't using the result. 
so the consumers are allowed to use both: 

```
function saveSearchSessionName () {
 await rename()
}
```
or
```
function saveSearchSessionName () {
  return rename()
}
```